account_invoice_ubl_mass_download/models/account_invoice.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2021 Coop IT Easy (https://coopiteasy.be)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import base64
+import io
+import logging
+import zipfile
+
+from odoo import api, models
+
+_logger = logging.getLogger(__name__)
+
+ATTACHMENT_TMP_NAME = "temporary-invoice-ubl-zip-file"
```

Comment:
Although unlikely with low traffic, but is it not possible that someone could receive the wrong attachment if two requests come in simultaneously?

Or in other words: Should the filename not be unique for each request? Append a random 7-character hash to this file name for every request, and in `_cron_remove_ubl_zip` remove all files that start with `ATTACHMENT_TMP_NAME`.